extensions/src/objectDetection/index.ts

```diff
@@ -0,0 +1,161 @@
+  detector: ObjectDetectorClass;
+
+  /**
+   * Tells whether the continuous detection should be on/off
+   */
+  continuous: boolean;
+
+  /**
+   * Dimensions of the video frame
+   */
+  DIMENSIONS = [480, 360];
+
+  /**
+   * The frequency at which the detector will generate new detections
+   */
+  processFreq: number;
+
+  /**
+   * Helper for creating ImageData objects (used to create drawables)
+   */
+  imageHelper: ReturnType<typeof getImageHelper>;
+
+  /**
+   * The list of drawables, which will all be displayed in the video frame
+   */
+  drawables: ReturnType<typeof this.createDrawable>[] = [];
+
+  /**
+   * The color of the detection box
+   */
+  color: string;
+
+  /**
+   * The thickness of the detection box
+   */
+  thickness: number;
+
+  /**
+   * Initializes the extension with standard values for the extension's class attributes.
+   */
+  async init() {
+    this.enableVideo()
+    this.continuous = false;
+    this.detector = await initializeObjectDetector();
+    this.imageHelper = getImageHelper(this.DIMENSIONS[0], this.DIMENSIONS[1])
+    this.color = 'white'
+    this.thickness = 5
+    this.processFreq = 100;
+  }
+
+  /**
+   * Runs a loop for constantly updating the video frame with a new set of detections.
+   * Used for the continuous detection block.
+   */
+  private async detectionLoop() {
+    while (this.continuous) {
+      const frame = this.getVideoFrame("canvas");
+      const start = Date.now();
+      if (frame) {
+        const detections = await this.detector.detect(frame);
+        this.displayImageDetections(detections);
+      }
+      const elapsed = Date.now() - start;
+      await untilTimePassed(this.processFreq - elapsed);
+      this.clearFrame()
+    }
+  }
+
+  /**
+   * Clears the frame of all drawables (resets the frame to blank video).
+   */
+  clearFrame() {
+    while (this.drawables.length > 0) this.drawables.shift().destroy();
+  }
+
+  /**
+   * Displays the detections from the detector in the video frame.
+   * @param detections The detections object returned from the detector's detect method
+   */
+  async displayImageDetections(detections) {
+    const { drawables, imageHelper } = this;
+    const rects = imageHelper.createRects(detections.detections, this.color, this.thickness)
+    drawables.push(this.createDrawable(rects));
+  }
+
+  @block({
+    type: "command",
+    text: (delay) => `Set detection rate to ${delay}`,
+    arg: {
+      type: "number",
+      options: [60, 30, 10, 2, 1],
+      defaultValue: 10
+    } as const
+  })
+  setFrameRate(delay: number) {
+    this.processFreq = 1000 / delay
+  }
+
+  @block({
+    type: "command",
+    text: (color) => `Set box color to ${color}`,
+    arg: "color"
+  })
+  setColor(color: RGBObject) {
+    this.color = rgbToHex(color);
+  }
+
+  @block({
+    type: "command",
+    text: (thickness) => `Set box thickness to ${thickness}`,
+    arg: {
+      type: "number",
+      defaultValue: 5,
+      handler: (x) => {
```

Comment:
Hm, is the `handler` actually doing? @pmalacho-mit to look

Reply:
A bit confused about what you're asking. The handler is supposed to limit the box thickness to a value between 0 and 20.

Reply:
@pmalacho-mit Looks like it isn't doing anything... Did I put the handler in the right spot?

Reply:
@jimbo797 ya, I think this is in the wrong spot. Did your editor not give you errors? Here's what it looks like for me:
<img width="482" alt="Screen Shot 2023-05-29 at 4 20 09 PM" src="https://github.com/mitmedialab/prg-extension-boilerplate/assets/95306112/e5dd7317-05d2-4985-a0d3-2a014db72622">

Reply:
You only have a handler when you have a menu that accepts reporters. So this would be correct (but note you'll need to deal with typescript errors because the input to the handler is of type `uknown` so you'll need to do some casting to make it happy):
<img width="644" alt="Screen Shot 2023-05-29 at 4 21 07 PM" src="https://github.com/mitmedialab/prg-extension-boilerplate/assets/95306112/6cab1d88-7452-4399-b9bd-4b3f56f73204">